### PR TITLE
Astar base UI initial commit

### PR DESCRIPTION
**Pull request by bobo-k2**
Created AstarBase basic UI with option to connect wallet and call minting contract.

There are some TODOs in the code which needs to be resolved.

### CODE REVIEW

**Comment by hoonsubin** [comment]
Screen Shot 2022-02-07 at 10 34 47 AM

Even though I'm connected to the page, it renders the connect button.
FYI the connect and mint button was switched on my machine.

**Comment by bobo-k2** [comment]
The button is displayed because you are connected to Astar and app expects you you to be on Shiden. I put TODO to implement automatic network switching. I can do this in the current branch too.

File: src/store/actions.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  if (networkId != chainConfig.network.id) {
+    const chainIdHex = `0x${chainConfig.network.id.toString(16)}`;
+
+    try {
+      await ethereum.request({
+        method: "wallet_switchEthereumChain",
+        params: [{ chainId: chainIdHex }]
+      });
+    } catch (error) {
+      if (error.code == UNRECOGNIZED_CHAIN_ID_ERROR_CODE ) {
```
Comment (hoonsubin) [inline src/store/actions.ts:29]:
```
(local var) error: unknown
Object is of type 'unknown'.ts(2571)
```

Type error

**Review by hoonsubin — CHANGES_REQUESTED** [review]
Can you also create a GH action for linting and building the code?

**Comment by bobo-k2** [comment]
Sure.

File: src/store/actions.ts (hunk 1 above)
Comment (bobo-k2) [reply src/store/actions.ts:29]: Weird. I can't reproduce. Let me check.

Comment (bobo-k2) [reply src/store/actions.ts:29]: It must have something to do with your type script version. I think I fixed the error. Please check again

**Review by hoonsubin — APPROVED** [review]
LGTM

**Review by impelcrypto — APPROVED** [review]
LGTM

**Comment by hoonsubin** [comment]
Closing this PR as @impelcrypto is working on the page.

**Comment by impelcrypto** [comment]
@hoonsubin 
Sorry, I meant I want to merge this PR first so I can make a new branch on top of this PR.

**Comment by hoonsubin** [comment]
> @hoonsubin Sorry, I meant I want to merge this PR first so I can make a new branch on top of this PR.

Ohh, my bad. I'll merge it now then

**Pull request merged**